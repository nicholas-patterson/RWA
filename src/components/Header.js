import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
  },
});

const Header = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar variant="dense">
          <Typography variant="h4" color="inherit" data-testid="site-title">
            React Weather App
          </Typography>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default Header;
