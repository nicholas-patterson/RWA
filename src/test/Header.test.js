import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Header from "../components/Header";

afterEach(cleanup);

describe("Header Component", () => {
  it("renders", () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("shows page main title", () => {
    const { getByTestId } = render(<Header />);

    expect(getByTestId("site-title")).toHaveTextContent(/React Weather App/);
  });
});
