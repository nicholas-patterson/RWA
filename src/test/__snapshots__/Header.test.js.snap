// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-colorInherit"
          data-testid="site-title"
        >
          React Weather App
        </h4>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
